main.py

```diff
@@ -47,35 +53,86 @@ def start_server(self, socket=os_socket):
                                 break
                             message = json.loads(data)
                             print()
-                            print(f"Received by {data}")
-                            logger.debug(f"Received by {message['sender']}: {message}")
-                            self.message_queue.put(data)
 
-                            if not self.message_queue.empty():
-                                msg = self.message_queue.get()
+                            if message.get("type") == "GET_NODES":
+                                conn.sendall(json.dumps({"nodes": self.peer_hosts}).encode())
+                                new_peer = message.get("node_ip")
```

Comment:
I think you should just look at the address that the request came from instead of the message content. That is simpler and more reliaible.
You can get the real ip address of the requestor from line
```
conn, addr = s.accept()
```
The addr should be the address of the peer